eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php

```diff
@@ -288,8 +288,28 @@ public function locationDeleted($locationId)
         $this->logger->logCall(__METHOD__, array('location' => $locationId));
         $return = $this->persistenceHandler->urlAliasHandler()->locationDeleted($locationId);
 
-        $this->cache->clear('urlAlias', 'location', $locationId);
+        $this->clearLocation($locationId);
 
         return $return;
     }
+
+    /**
+     * @param $locationId
+     */
+    protected function clearLocation($locationId)
+    {
+        $locationCache = $this->cache->getItem('urlAlias', 'location', $locationId);
+
+        if ($locationCache->isMiss()) {
+            // we need to clear all if we don't have location id in cache
+            $this->cache->clear('urlAlias');
+        } else {
+            $urlAliasesId = $locationCache->get();
+
+            foreach ((array)$urlAliasesId as $eachUrlAliasId) {
```

Comment:
@andrerom 

So as I excepted, in unit test there are places where the get method returns for example null or 0. And this can not be used in foreach, so if I should remove the array cast, what is the best next step? 

1. Change this test and mock it to return array?
2. Add condition for run only if the given value is an array?
3. Cast the value to array?